json-schema-core/src/main/java/io/ballerina/jsonschema/core/Generator.java

```diff
@@ -0,0 +1,254 @@
+        for (Object element : enumKeyword) {
+            if (typeList.contains(element.getClass())) {
+                finalList.add(element);
+            }
+        }
+
+        if (constKeyword == null) {
+            return new ArrayList<>(finalList);
+        }
+
+        if (finalList.contains(constKeyword)) { // Checked for reference cross check-ins
+            return new ArrayList<>(List.of(constKeyword));
+        }
+        return new ArrayList<>();
+    }
+
+    public static String getBallerinaType(Object type) {
+        if (type == Long.class) {
+            return "Integer";
+        }
+        if (type == Double.class) {
+            return "Number";
+        }
+        if (type == String.class) {
+            return "String";
+        }
+        if (type == Boolean.class) {
+            return "Boolean"; // Redundant
+        }
+        if (type == ArrayList.class) {
+            return "Array";
+        }
+        if (type == LinkedHashMap.class) {
+            return "Object";
+        }
+        return "Null";
+    }
+
+    public ModulePartNode generateModulePartNode() throws Exception {
+        NodeList<ModuleMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createNodeList(this.nodes.values());
+        NodeList<ImportDeclarationNode> imports = getImportDeclarations();
+        Token eofToken = AbstractNodeFactory.createIdentifierToken(EOF_TOKEN);
+        return NodeFactory.createModulePartNode(imports, moduleMembers, eofToken);
+    }
+
+    public String formatModuleParts(ModulePartNode modulePartNode) throws FormatterException {
+        ForceFormattingOptions forceFormattingOptions = ForceFormattingOptions.builder()
+                .setForceFormatRecordFields(true).build();
+        FormattingOptions formattingOptions = FormattingOptions.builder()
+                .setForceFormattingOptions(forceFormattingOptions).build();
+        return Formatter.format(modulePartNode.syntaxTree(), formattingOptions).toSourceCode();
+    }
+
+    public NodeList<ImportDeclarationNode> getImportDeclarations() throws Exception {
+        Collection<ImportDeclarationNode> imports = new ArrayList<>();
+        for (String module : this.getImports()) {
+            ImportDeclarationNode node = NodeParser.parseImportDeclaration(module);
+            if (node.hasDiagnostics()) {
+                throw new Exception(INVALID_IMPORTS_ERROR);
+            }
+            imports.add(node);
+        }
+        return AbstractNodeFactory.createNodeList(imports);
+    }
+
+    public void addImports(String module) {
+        String importDeclaration = IMPORT + WHITE_SPACE + module + SEMI_COLON;
+        if (!this.imports.contains(importDeclaration)) {
+            this.imports.add(importDeclaration);
+        }
+    }
+
+    public void addDiagnostic(JsonSchemaDiagnostic diagnostic) {
+        this.diagnostics.add(diagnostic);
+    }
+
+    public ArrayList<String> getImports() {
+        return new ArrayList<>(this.imports);
+    }
+}
```

Comment:
Is it ok to overwrite the complete existing `types.bal`

Reply:
If a file already exists, we have to ask the user whether to overwrite like we do in the OpenAPI tool.